src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexCompiler.cs

```diff
@@ -1453,27 +1453,27 @@ internal void GenerateFindFirstChar()
                 else
                     Rightcharnext();
 
-                if (_fcPrefix.CaseInsensitive)
+                if (_fcPrefix.Value.CaseInsensitive)
```

Comment:
Do we know at all of these places where Value is being accessed that it's non-null?  If so, GetValueOrDefault() is slightly faster, as it doesn't involve a null check, whereas Value does (throwing if it's null).

Reply:
yes, everywhere where I added `Value.` it's safe to assume that the value is never null. Thanks, didn't know that.